src/display/mapcanvas.cpp

```diff
@@ -1490,6 +1490,50 @@ void MapCanvas::drawRoomDoorName(const Room *sourceRoom, uint sourceDir, const R
     qint32 dX = srcX - tarX;
     qint32 dY = srcY - tarY;
 
+    // Look at door flags to code postfixes
+    DoorFlags sourceDoorFlags = getDoorFlags(sourceRoom->exit(sourceDir));
+    QString sourcePostFix;
+    if (ISSET(sourceDoorFlags, DF_HIDDEN))
+    {
+      sourcePostFix = "h";
+    }
+    if (ISSET(sourceDoorFlags, DF_NEEDKEY))
+    {
+      sourcePostFix += "L";
+    }
+    if (ISSET(sourceDoorFlags, DF_NOPICK))
+    {
+      sourcePostFix += "/NP";
+    }
+    if (ISSET(sourceDoorFlags, DF_DELAYED))
+    {
+      sourcePostFix += "d";
+    }
+    if (sourcePostFix.length() > 0) {
+      sourcePostFix = " [" + sourcePostFix + "]";
+    }
+    DoorFlags targetDoorFlags = getDoorFlags(targetRoom->exit(targetDir));
+    QString targetPostFix;
+    if (ISSET(targetDoorFlags, DF_HIDDEN))
+    {
+      targetPostFix = "h";
+    }
+    if (ISSET(targetDoorFlags, DF_NEEDKEY))
+    {
+      targetPostFix += "L";
+    }
+    if (ISSET(targetDoorFlags, DF_NOPICK))
+    {
+      targetPostFix += "/NP";
```

Comment:
Why add a "/" in front of this combination and not others?

Reply:
I started thinking about this again. Maybe instead of using /NP we can add ANSI color codes to the "L" in the lock? Perhaps red? :-)